### ISSUE TITLE
Running citar-create-note doesn't add a "#+reference"

### ISSUE THREAD

**Issue by psmith1303** [pprevos/citar-denote#14]
I'm in the process of rig-jigging my (academic) workflow, and am cleaning up my emacs.d configuration.

That's seen me adopt `vertico `(1.1), `denote `(1.2.0), and `citar` (20230227.2247). The `citar-denote` package (20230211.528) looks like it should be just the thing I need to join citar and denote.

I've installed it thus:

> 
> (use-package citar-denote
>   :after citar denote
>   :config
>   (citar-denote-mode)
>   (setq citar-open-always-create-notes t))
> 
> 

Alas ... there's a wee problem, when I create a new note (`citar-create-note`) a new buffer is created that with the right frontmatter (a title, date, filetags, and identifier), but without a reference.

 E.g., a buffer called "20230305T155127--organizational-strategy-structure-and-process__bib_strategy.org" is created; this looks correct to me and has the following content.

> 
> #+title:      Organizational Strategy, Structure, and Process
> #+date:       [2023-03-05 Sun 15:51]
> #+filetags:   :bib:strategy:
> #+identifier: 20230305T155127
> 

But I would expect to see a reference like "`#+reference:  miles_organizational_1978`"

There is an error message that says, "citar-denote-add-reference: Buffer is not a Denote file". 

Looking at the mode-line, it says `Org citar-denote Ind Fly/en`, which is exactly what it says when I open any other denote file.

I'm a little lost now ... any ideas

**Comment by psmith1303** [comment]
A bit more info: Looking at my messages buffer, it says "citar-denote-add-reference: Buffer is not a Denote file". So, I guess that's the function where the problem lies.

**Comment by pprevos** [comment]
Hi Peter, apologies for the package not working.  I fixed some issues but had not committed it to GitHub just yet. I have just pushed a new version to GitHub. Can you test it please?

My config:

```
  (use-package citar-denote
    :ensure nil
    :load-path "~/Documents/emacs/citar-denote" ;; Location where you store this version
    :config
    (citar-denote-mode)
    :custom
    (citar-open-always-create-notes t)
    :bind (("C-c d c c" . citar-create-note)
           ("C-c d c a" . citar-denote-add-citekey)
           ("C-c d c x" . citar-denote-remove-citekey)
           ("C-c d c o" . citar-denote-open-note)
           ("C-c d c d" . citar-denote-dwim)
           ("C-c d c r" . citar-denote-find-reference)
           ("C-c d c f" . citar-denote-find-citation)
           ("C-c d c n" . citar-denote-find-nocite)))
```

I tested every function with a fresh Emacs session and they all seem to work.

Hope you get it going. Adding references to Denote has been a great tool for my own research. 

I see you are reading Miles and Snow. It is amazing how so much of business theory can seemingly be explained in a two-dimensional matrix :)

**Comment by pprevos** [comment]
The latest version has now also hit MELPA. Is it working for you now?

**Comment by psmith1303** [comment]
Hello Peter,

On Mon, 6 Mar 2023, at 09:41, Peter Prevos wrote:
> The latest version has now also hit MELPA. Is it working for you now?
> 
Yes, that seems to work.

I deleted the package, closed and opened emacs, and reinstalled the package from MELPA.

I've checked, and `citar-create-note` now works as expected. Thanks.

I can now get back to rebuilding/remodelling my notes and maybe integrating them with my website.


Regards, Peter


p.s. Next time you're in Auckland, look me up.
